bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/Utils.java

```diff
@@ -66,6 +93,65 @@ static Object getTomlTypeString(TomlNode tomlNode) {
         }
     }
 
+    static Object getBalValue(TomlNode tomlNode, Set<TomlNode> visitedNodes) {
```

Comment:
If this method is only used in the `TomlProvider` class, wouldn't it be better to have it in that class itself?

Reply:
Actually, all methods in this class used only by TomlProvider. I think util methods are moved to here. should be fine I think @HindujaB

Reply:
Yes. We have only the util methods in this class.
As per best practice, +1 to move the methods to `TomlProvider`. But will removing this class can cause any cache issues like `ConfigurationMapper`?

Reply:
Lets keep this since tomlProvider class already have too many methods.